### PR TITLE
chore: migrate media-library initial setup

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

- Migrate media-library plugin admin to Typescript
  - fix some Type errors we had in the BE unit tests
  - define the basic TS configuration for the admin part

based on the work made in the draft PR https://github.com/strapi/strapi/pull/20867

### Files to migrate?

- [ ] admin
  - [ ] src
    - [ ] components
      - [ ] AssetCard
        - [ ] tests
          - [ ] AssetCardBase.test.jsx
          - [ ] DocAssetCard.test.jsx
          - [ ] ImageAssetCard.test.jsx 
        - [ ] AssetCard.jsx
        - [ ] AssetCardBase.jsx
        - [ ] AudioAssetCard.jsx
        - [ ] AudioPreview.jsx
        - [ ] DocAssetCard.jsx
        - [ ] ImageAssetCard.jsx
        - [ ] UploadingAssetCard.jsx
        - [ ] VideoAssetCard.jsx
        - [ ] VideoPreview.jsx 
      - [ ] AssetDialog
        - [ ] BrowseStep
          - [ ] PaginationFooter
            - [ ] components.jsx
            - [ ] index.jsx
            - [ ] Pagination.jsx
            - [ ] PaginationContext.jsx  
        - [ ] SearchAsset
          - [ ] tests
            - [ ] index.test.jsx
          - [ ] index.jsx
        - [ ] tests
          - [ ] __snapshots__
          - [ ] index.test.jsx
        - [ ] utils
          - [ ] tests
            - [ ] isSelectable.test.js
          - [ ] isSelectable.js     
        - [ ] Filters.jsx
        - [ ] index.jsx
        - [ ] PageSize.jsx 
        - [ ] SelectedStep
          - [ ] index.jsx
        - [ ] tests
          - [ ] AssetDialog.test.jsx
        - [ ] DialogFooter.jsx
        - [ ] index.jsx  
      - [ ] AssetGridList
        - [ ] tests
          - [ ] __snapshots__
          - [ ] AssetGridList.test.jsx 
        - [ ] Draggable.jsx
        - [ ] index.jsx 
      - [ ] Breadcrumbs
        - [ ] tests
          - [ ] __snapshots__
          - [ ] index.test.jsx 
        - [ ] Breadcrumbs.jsx
        - [ ] CrumbSimpleMenuAsync.jsx
        - [ ] index.jsx
      - [ ] BulkMoveDialog
        - [ ] tests
          - [ ] __snapshots__
          - [ ] BulkMoveDialog.test.jsx 
        - [ ] BulkMoveDialog.jsx
        - [ ] index.jsx 
      - [ ] ContextInfo
        - [ ] tests
          - [ ] __snapshots__
          - [ ] ContextInfo.test.jsx 
        - [ ] ContextInfo.jsx
        - [ ] index.jsx 
      - [ ] CopyLinkButton
        - [ ] index.jsx 
      - [ ] EditAssetDialog
        - [ ] PreviewBox
          - [ ] AssetPreview.jsx
          - [ ] components.jsx
          - [ ] CroppingActions.jsx
          - [ ] index.jsx
        - [ ] tests
          - [ ] __snapshots__
          - [ ] EditAssetDialog.test.jsx
          - [ ] index.test.jsx
          - [ ] RemoveAssetDialog.test.jsx
        - [ ] DialogHeader.jsx
        - [ ] index.jsx
        - [ ] RemoveAssetDialog.jsx
        - [ ] ReplaceMediaButton.jsx     
      - [ ] EditFolderDialog
        - [ ] ModalHeader
          - [ ] index.jsx
          - [ ] ModalHeader.jsx
        - [ ] tests
          - [ ]  __snapshots__
          - [ ] EditFolderDialog.test.jsx
        - [ ] EditFolderDialog.jsx
        - [ ] index.jsx
        - [ ] RemoveFolderDialog.jsx
      - [ ] EmptyAssets
        - [ ] EmptyAssetGrid.jsx
        - [ ] index.jsx
      - [ ] FilterList
        - [ ] tests
          - [ ]  __snapshots__
          - [ ] index.test.js
        - [ ] FilterTag.jsx
        - [ ] index.jsx  
      - [ ] FilterPopover
        - [ ] utils
          - [ ] getFilterList.js
        - [ ] FilterValueInput.jsx
        - [ ] index.jsx 
      - [ ] FolderCard
        - [ ] contexts
        - [ ] FolderCard
          - [ ] FolderCard.jsx
          - [ ] index.jsx 
        - [ ] FolderCardBody
          - [ ] FolderCardBody.jsx
          - [ ] index.jsx
        - [ ] FolderCardBodyAction
          - [ ] index.js 
        - [ ] FolderCardCheckbox
          - [ ] FolderCardCheckbox.jsx
          - [ ] index.jsx
        - [ ] hooks
          - [ ] tests
            - [ ] useId.test.js
          - [ ] useId.js
        - [ ] tests
          - [ ] __snapshots__
          - [ ] FolderCard.test.jsx 
        - [ ] index.jsx
      - [ ] FolderGridList
        - [ ] tests
          - [ ] __snapshots__
          - [ ] FolderGridList.test.jsx
        - [ ] FolderGridList.jsx
        - [ ] index.jsx 
      - [ ] MediaLibraryDialog
        - [ ] index.jsx 
      - [ ] MediaLibraryInput
        - [ ] Carousel
          - [ ] tests
            - [ ] CarouselAssets.test.jsx
          - [ ] CarouselAsset.jsx
          - [ ] CarouselAssetActions.jsx
          - [ ] CarouselAssets.jsx
          - [ ] EmptyStateAsset.jsx 
        - [ ] tests
          - [ ] MediaLibraryInput.test.jsx 
        - [ ] index.jsx
      - [ ] SelectTree
        - [ ] tests
          - [ ] __snapshots__
          - [ ] SelectTree.test.jsx
        - [ ] utils
          - [ ] tests
            - [ ] __snapshots__
            - [ ] flattenTree.test.js
            - [ ] getOpenValues.test.js
            - [ ] getValuesToClose.test.js
          - [ ] flattenTree.js
          - [ ] getOpenValues.js
          - [ ] getValuesToClose.js
        - [ ] index.jsx
        - [ ] Option.jsx
        - [ ] SelectTree.jsx
        - [ ] SelectTree.stories.mdx 
      - [ ] SortPicker
        - [ ] index.jsx
      - [ ] TableList
        - [ ] tests
          - [ ] __snapshots__
          - [ ] CellContent.test.jsx
          - [ ] TableList.test.jsx
          - [ ] TableRows.test.jsx
        - [ ] CellContent.jsx
        - [ ] index.jsx
        - [ ] PreviewCell.jsx
        - [ ] TableRows.jsx
      - [ ] UploadAssetDialog
        - [ ] AddAssetStep
          - [ ] tests
            - [ ] __snapshots__
            - [ ] FromComputerForm.test.jsx
          - [ ] AddAssetStep.jsx
          - [ ] FromComputerForm.jsx
          - [ ] FromUrlForm.jsx 
        - [ ] PendingAssetStep
          - [ ] tests
            - [ ] __snapshots__
            - [ ] PendingAssetStep.test.jsx
          - [ ] PendingAssetStep.jsx 
        - [ ] tests
          - [ ] UploadAssetDialog.test.jsx
        - [ ] UploadAssetDialog.jsx 
      - [ ] UploadProgress
        - [ ] tests
          - [ ] index.test.jsx 
        - [ ] index.jsx
    - [ ] hooks
      - [ ] __mocks
        - [ ] useAssets.js
        - [ ] useBulkMove.js
        - [ ] useBulkRemove.js
        - [ ] useConfig.js
        - [ ] useEditFolder.js
        - [ ] useFolder.js
        - [ ] useFolders.js
        - [ ] useFolderStructure.js
        - [ ] useMediaLibraryPermissions.js
        - [ ] useModalQueryParams.js 
      - [ ] tests
        - [ ] useAssets.test.js
        - [ ] useBulkMove.test.js
        - [ ] useBulkRemove.test.js
        - [ ] useClipboard.test.ts
        - [ ] useConfig.test.js
        - [ ] useEditFolder.test.js
        - [ ] useFolder.test.js
        - [ ] useFolders.test.js
        - [ ] useFolderStructure.test.js
        - [ ] useModalQueryParams.test.js
        - [ ] usePersistentState.test.ts
        - [ ] useRemoveAsset.test.js
        - [ ] useSelectionState.test.ts 
      - [ ] utils
        - [ ] tests
          - [ ] rename-keys.test.js 
        - [ ] rename-keys.js
      - [ ] useAssets.js
      - [ ] useBulkMove.js
      - [ ] useBulkRemove.js
      - [ ] useClipboard.ts
      - [ ] useConfig.js
      - [ ] useCropImg.js
      - [ ] useEditAsset.js
      - [ ] useEditFolder.js
      - [ ] useFolder.js
      - [ ] useFolders.js
      - [ ] useFolderStructure.js
      - [ ] useMediaLibraryPermissions.js
      - [ ] useModalQueryParams.js
      - [ ] usePersistentState.ts
      - [ ] useRemoveAsset.js
      - [ ] useSelectionState.ts
      - [ ] useUpload.js 
    - [ ] pages
      - [ ] App
        - [ ] components
          - [ ] tests
            - [ ] EmptyOrNoPermissions.test.jsx
          - [ ] BulkActions.jsx
          - [ ] BulkDelete.jsx
          - [ ] BulkMoveButton.jsx
          - [ ] EmptyOrNoPermissions.jsx
          - [ ] Filters.jsx
          - [ ] Header.jsx
        - [ ] ConfigureTheView
          - [ ] components
            - [ ] tests
              - [ ] __snapshots 
            - [ ] Settings.jsx
          - [ ] state
            - [ ] tests
              - [ ] init.test.js
              - [ ] reducer.test.js
            - [ ] actions.js
            - [ ] actionTypes.js
            - [ ] init.js
            - [ ] reducer.js 
          - [ ] tests
            - [ ] __snapshots
            - [ ] ConfigureTheView.test.jsx
          - [ ] index.jsx
        - [ ] MediaLibrary
          - [ ] components
            - [ ] tests
              - [ ] __snapshots__
              - [ ] BulkActions.test.jsx
              - [ ] BulkDeleteButton.test.jsx
              - [ ] BulkMoveButton.test.jsx
              - [ ] EmptyOrNoPermissions.test.jsx
              - [ ] Header.test.jsx
            - [ ] BulkActions.jsx
            - [ ] BulkDeleteButton.jsx
            - [ ] BulkMoveButton.jsx
            - [ ] EmptyOrNoPermissions.jsx
            - [ ] Filters.jsx
            - [ ] Header.jsx
          - [ ] tests
            - [ ] MediaLibrary.test.jsx
          - [ ] index.jsx
        - [ ] index.jsx 
      - [ ] SettingsPage
        - [ ] tests
          - [ ] index.test.jsx
          - [ ] reducer.test.js 
        - [ ] index.jsx
        - [ ] init.js
        - [ ] reducer.js
    - [ ] translations
      - [ ] tests
        - [ ] plural.test.js
    - [ ] utils
      - [ ] tests
        - [ ] appendSearchParamsToUrl.test.js
        - [ ] containsAssetFilter.test.js
        - [ ] downloadFile.test.js
        - [ ] findRecursiveFolderByValue.test.js
        - [ ] findRecursiveFolderMetadatas.test.js
        - [ ] formatBytes.test.js
        - [ ] formatDuration.test.js
        - [ ] getAllowedFiles.test.js
        - [ ] getAPIInnerErrors.test.js
        - [ ] getBreadcrumbDataCM.test.js
        - [ ] getBreadcrumbDataML.test.js
        - [ ] getFileExtension.test.js
        - [ ] getFolderParents.test.js
        - [ ] getFolderURL.test.js
        - [ ] normalizeAPIError.test.js
        - [ ] prefixFileUrlWithBackendUrl.test.js
        - [ ] prefixPluginTranslations.test.js
        - [ ] toSingularTypes.test.js
        - [ ] typeFromMime.test.js
      - [ ] appendSearchParamsToUrl.js
      - [ ] containsAssetFilter.js
      - [ ] createAssetUrl.js
      - [ ] displayedFilters.js
      - [ ] downloadFilte.js
      - [ ] findRecursiveFolderByValue.js
      - [ ] findRecursiveFolderMetadatas.js
      - [ ] formatBytes.js
      - [ ] formatDuration.js
      - [ ] getAllowedFiles.js
      - [ ] getAPIInnerErrors.js
      - [ ] getBreadcrumbDataCM.js
      - [ ] getBreadcrumbDataML.js
      - [ ] getFileExtension.js
      - [ ] getFolderParents.js
      - [ ] getFolderURL.js
      - [ ] getTrad.js
      - [ ] index.js
      - [ ] moveElement.js
      - [ ] normalizeAPIError.js
      - [ ] prefixFileUrlWithBackendUrl.js
      - [ ] prefixPluginTranslations.js
      - [ ] rawFileToAsset.js
      - [ ] toSingularTypes.js
      - [ ] typeFromMime.js
      - [ ] urlsToAssets.js
      - [ ] urlYupSchema.js
    - [ ] constants.js
    - [ ] index.js
    - [ ] pluginId.js 
  - [x] tests  
    - [x] handlers.js
    - [x] setup.js
    - [x] utils.jsx 
- [ ] documentation 
  - [ ] content-api.json
- [x] .eslintignore
- [x] .eslintrc
- [x] .gitattributes
- [x] .gitignore
- [x] .npmignore
- [x] jest.config.front.js
- [x] jest.config.js
- [x] jsconfig.json
- [x] LICENSE
- [ ] package.json
- [ ] packup.config.ts
- [x] README.md
- [x] strapi-server.js 

Define the 
 - [ ] Contracts

### CODE REVIEW

File: packages/core/upload/server/src/controllers/validation/admin/folder-file.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ const validateStructureMoveManyFoldersFilesSchema = yup
 
 const validateDuplicatesMoveManyFoldersFilesSchema = yup
   .object()
+  // eslint-disable-next-line func-names
   .test('are-folders-unique', 'some folders already exist', async function (value) {
```
Comment (remidej) [inline packages/core/upload/server/src/controllers/validation/admin/folder-file.ts:33]: With an arrow function this wouldn't be required

File: packages/core/upload/server/src/controllers/validation/admin/folder-file.ts [hunk 2]
```diff
@@ -53,40 +54,39 @@ const validateDuplicatesMoveManyFoldersFilesSchema = yup
...
-
-      const destinationFolder = await strapi.db.query(FOLDER_MODEL_UID).findOne({
-        select: ['path'],
-        where: { id: destinationFolderId },
-      });
+const validateMoveFoldersNotInsideThemselvesSchema = yup.object().test(
+  'dont-move-inside-self',
+  'folders cannot be moved inside themselves or one of its children',
+  // eslint-disable-next-line func-names
+  async function (value) {
```
Comment (remidej) [inline packages/core/upload/server/src/controllers/validation/admin/folder-file.ts:61]: same thing about arrow function

File: packages/core/upload/admin/src/pages/App/ConfigureTheView/state/reducer.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import set from 'lodash/set';
 import { ON_CHANGE, SET_LOADED } from './actionTypes';
 import { init, initialState } from './init';
 
+// eslint-disable-next-line default-param-last
 const reducer = (state = initialState, action) =>
```
Comment (remidej) [inline packages/core/upload/admin/src/pages/App/ConfigureTheView/state/reducer.js:9]: https://github.com/strapi/strapi/pull/21097#discussion_r1766562881

Comment (simotae14) [reply packages/core/upload/admin/src/pages/App/ConfigureTheView/state/reducer.js]: I added a new default action

File: packages/core/upload/server/src/controllers/validation/admin/folder-file.ts (hunk 1 above)
Comment (simotae14) [reply packages/core/upload/server/src/controllers/validation/admin/folder-file.ts]: I preferred to solve the problem by giving a name to the function

File: packages/core/upload/server/src/controllers/validation/admin/folder-file.ts (hunk 2 above)
Comment (simotae14) [reply packages/core/upload/server/src/controllers/validation/admin/folder-file.ts]: I preferred to solve the problem by giving a name to the function

**Review by remidej — APPROVED** [review]

**Review by remidej — DISMISSED** [review]
Working as expected 👍

**Review by remidej — APPROVED** [review]

**Pull request merged**